lib/hash/hash_test.go

```diff
@@ -0,0 +1,66 @@
+package hash_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/syntasso/kratix/lib/hash"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+)
+
+var _ = Describe("Hasher", func() {
+	var (
+		input *unstructured.Unstructured
+	)
+
+	BeforeEach(func() {
+		input = &unstructured.Unstructured{
+			Object: map[string]interface{}{
+				"apiVersion": "v1",
+				"kind":       "Pod",
+				"metadata": map[string]interface{}{
+					"name":      "test-pod",
+					"namespace": "test-namespace",
+				},
+				"spec": map[string]interface{}{
+					"foo": "bar",
+				},
+				"status": map[string]interface{}{
+					"some": "status",
+				},
+			},
+		}
+	})
+
+	Describe("ComputeHash", func() {
+		const expectedHash = "9bb58f26192e4ba00f01e2e7b136bbd8"
+
+		It("hashes the inpuy Object Spec", func() {
+			actualHash, err := hash.ComputeHash(input)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(actualHash).To(Equal(expectedHash))
+		})
+
+		It("remains constant if the request spec hasn't changed", func() {
```

Comment:
worth adding labels to the things we trigger changes on? maybe overkill now but could see it in the future

Reply:
you mean adding the spec diff? 🤔

Reply:
I mean if someone adds a new metadata.label as the only change to a RR it may be worth retriggering the pipeline

Reply:
gotcha!

i thought about it, and you are right, i can see as something we will potentially need. however, at the same time, keeping it spec only simplifies the code a bit -- and we can always add triggering on other things in the future?

Reply:
sounds reasonable 💯